src/nvme/tree.c

```diff
@@ -1220,6 +1220,126 @@ void nvme_ctrl_set_tls_key(nvme_ctrl_t c, const char *key)
 		c->tls_key = strdup(key);
 }
 
+void nvme_ctrl_set_addr(nvme_ctrl_t c, const char *addr)
+{
+	if (c->address) {
+		free(c->address);
```

Comment:
`free()` is NULL-safe. So you could just call it w/o first cheking that the pointer in non-NULL.

https://man7.org/linux/man-pages/man3/free.3p.html